server/app/helpers/wait_helper.rb

```diff
@@ -0,0 +1,51 @@
+module WaitHelper
+  include Logging
+
+  WAIT_TIMEOUT = 300
+  WAIT_INTERVAL = 0.5
+
+  def _wait_now
+    Time.now.to_f
+  end
+
+  # Wait until given block returns truthy value, returning nil on timeout
+  #
+  # @param timeout [Fixnum] How long to wait
+  # @param interval [Fixnum] At what interval is the block yielded
+  # @param message [String] Message for debugging
+  # @param block [Block] Block to yield
+  # @return [Object] Return value from block, or nil
+  def wait_until(timeout: WAIT_TIMEOUT, interval: WAIT_INTERVAL, message: nil, &block)
+    raise ArgumentError, 'no block given' unless block_given?
+
+    wait_until = _wait_now + timeout
+
+    loop do
+      return nil if _wait_now > wait_until
+
+      value = yield
+
+      if value
+        return value
+      else
+        debug "wait... #{message}"
+        sleep interval
+      end
+    end
+  end
+
+  # Wait until given block returns truthy value, raising on timeout
+  #
+  # @return [Object] Last return value of the block
+  # @raise [Timeout::Error] If block does not return truthy value within given timeout
+  def wait_until!(**opts, &block)
+    if value = wait_until(**opts, &block)
+      return value
+    else
+      raise Timeout::Error
```

Comment:
Would it make sense to put in some "context" for the error? Now the logs don't show much on these:
```
E, [2017-03-16T09:45:47.758033 #16] ERROR -- GridServiceInstanceDeployer: Timeout::Error
api_1      | E, [2017-03-16T09:45:47.758132 #16] ERROR -- GridServiceInstanceDeployer: /app/app/helpers/wait_helper.rb:45:in `wait_until!'
api_1      | /app/app/services/grid_service_instance_deployer.rb:36:in `wait_for_service_to_start'
api_1      | /app/app/services/grid_service_instance_deployer.rb:22:in `deploy'
api_1      | /app/app/services/grid_service_deployer.rb:112:in `block in deploy_service_instance'
api_1      | /usr/lib/ruby/gems/2.3.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
```
It would help debugging if there would be something like:
```
E, [2017-03-16T09:45:47.758033 #16] ERROR -- GridServiceInstanceDeployer: Timeout::Error: service startup
```